_pylibczi/_pylibczi.cpp

```diff
@@ -207,6 +229,104 @@ static PyObject *cziread_allsubblocks(PyObject *self, PyObject *args) {
     return Py_BuildValue("OO", images, (PyObject *) coordinates);
 }
 
+// Begin new code
+
+ static int convertDictToPlaneCoords(PyObject *obj, void * dim_p){
+    if(!PyDict_Check(obj)){ // docs says it returns true/false but it returns an integer
+        return 0; // not a dictionary somethings wrong
+    };
+
+    libCZI::CDimCoordinate *dims = static_cast<libCZI::CDimCoordinate *>(dim_p);
+    std::map < std::string, libCZI::DimensionIndex > tbl{
+            {"V", libCZI::DimensionIndex::V},
+            {"H", libCZI::DimensionIndex::H},
+            {"I", libCZI::DimensionIndex::I},
+            {"S", libCZI::DimensionIndex::S},
+            {"R", libCZI::DimensionIndex::R},
+            {"T", libCZI::DimensionIndex::T},
+            {"C", libCZI::DimensionIndex::C},
+            {"Z", libCZI::DimensionIndex::Z}
+    };
+    std::for_each(tbl.begin(), tbl.end(), [&](std::pair< const std::string, libCZI::DimensionIndex > &kv){
+        PyObject *pyLetter;
+        PyObject *pyInt;
+        pyLetter = Py_BuildValue("s",kv.first.c_str());
+        if(PyDict_Contains(obj, pyLetter)){
+            int dimValue = 0;
+            pyInt = PyDict_GetItem(obj, pyLetter);
+            dimValue = static_cast<int>(PyLong_AsLong(pyInt));
+            dims->Set(kv.second, dimValue);
+        }
+        Py_DecRef(pyLetter);
+    });
+    return 1; // success
+}
+
+static int listToIntRect(PyObject *obj, void * rect_p){
+    libCZI::IntRect *rect = static_cast<libCZI::IntRect *>(rect_p);
+    if( obj == Py_None ){
+        rect->x = rect->y = rect->w = rect->h = 0;
+        return 1;
+    }
+
+    if(!PyTuple_Check(obj)){
+        std::cout << "region not a Tuple!" << std::endl;
+    }
+
+    if(PyTuple_Size(obj) != 4){
+        std::cerr << "Error: format for region (x, y, w, h)! " << std::endl;
+        return 0;
+    }
+
+    rect->x = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 0)));
```

Comment:
What is the result if the item in the tuple is not an integer? Am I remembering correctly that PyLong_AsLong will return -1?

Reply:
It returns -1 but it sets a python error. I modified the code to check for the python error and set an additional Python error message. Thanks for the suggestion